lightning-rapid-gossip-sync/src/processing.rs

```diff
@@ -169,23 +169,15 @@ impl<NG: Deref<Target=NetworkGraph<L>>, L: Deref> RapidGossipSync<NG, L> where L
 			if (channel_flags & 0b_1000_0000) != 0 {
 				// incremental update, field flags will indicate mutated values
 				let read_only_network_graph = network_graph.read_only();
-				if let Some(channel) = read_only_network_graph
-					.channels()
-					.get(&short_channel_id) {
-
-					let directional_info = channel
-						.get_directional_info(channel_flags)
-						.ok_or(LightningError {
-							err: "Couldn't find previous directional data for update".to_owned(),
-							action: ErrorAction::IgnoreError,
-						})?;
-
+				if let Some(directional_info) =
+					read_only_network_graph.channels().get(&short_channel_id)
+					.map(|channel| channel.get_directional_info(channel_flags)).unwrap_or(None)
```

Comment:
doing a `map` and then a `unwrap_or(None)` can just be simplified with a `and_then`